tests/api/plugins/content-releases/releases.test.api.ts

```diff
@@ -0,0 +1,646 @@
+
+    test('cannot create a scheduled release with date in the past', async () => {
+      const res = await rq({
+        method: 'POST',
+        url: '/content-releases/',
+        body: {
+          name: 'Test Scheduled Release',
+          scheduledAt: '2022-10-10T00:00:00.000Z',
+          isScheduled: true,
+          time: '00:00',
+          date: '2022-01-01',
+          timezone: 'Europe/Madrid',
+        },
+      });
+
+      expect(res.statusCode).toBe(400);
+      expect(res.body.error.message).toBe('Scheduled at must be later than now');
+    });
+  });
+
+  // Depends on the previous test
+  describe('Create Release Actions', () => {
+    test('Create a release action with valid status', async () => {
+      const release = data.releases[0];
+
+      const res = await rq({
+        method: 'POST',
+        url: `/content-releases/${release.id}/actions`,
+        body: {
+          entry: {
+            id: 1,
+            contentType: productUID,
+          },
+          type: 'publish',
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body.data.isEntryValid).toBe(true);
+    });
+
+    test('Create a release action with invalid status', async () => {
+      const release = data.releases[0];
+
+      const res = await rq({
+        method: 'POST',
+        url: `/content-releases/${release.id}/actions`,
+        body: {
+          entry: {
+            id: 2,
+            contentType: productUID,
+          },
+          type: 'unpublish',
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body.data.isEntryValid).toBe(false);
+    });
+
+    test('cannot create an action with invalid contentTypeUid', async () => {
+      const release = data.releases[0];
+
+      const res = await rq({
+        method: 'POST',
+        url: `/content-releases/${release.id}/actions`,
+        body: {
+          entry: {
+            id: 1,
+            contentType: 'invalid',
+          },
+          type: 'publish',
+        },
+      });
+
+      expect(res.statusCode).toBe(404);
+      expect(res.body.error.message).toBe('No content type found for uid invalid');
+    });
+
+    test('throws an error when trying to add an entry that is already in the release', async () => {
+      const release = data.releases[0];
+
+      const res = await rq({
+        method: 'POST',
+        url: `/content-releases/${release.id}/actions`,
+        body: {
+          entry: {
+            id: 1,
+            contentType: productUID,
+          },
+          type: 'publish',
+        },
+      });
+
+      expect(res.statusCode).toBe(400);
+      expect(res.body.error.message).toBe(
+        'Entry with id 1 and contentType api::product.product already exists in release with id 1'
+      );
+    });
+  });
+
+  // Depends on the previous test
+  describe('Create Many Release Actions', () => {
+    test('Create many release actions', async () => {
+      const release = data.releases[0];
```

Comment:
We are using this pattern in many places in strapi api tests (tests relying on other tests pushing into a data object)

Not requesting to change this , just that I think it's a safer approach to not rely on other tests, as it requires user reading the tests to look to another place to understand what's happening.

Someone could add a test that modifies that data, and break downstream tests.

As an alternative (not asking to change it :D ) even if it's a bit more verbose I would do something like:

```ts
const { release } = await createRelease()
await createPublishAction(release.id, publishActionPayload(1))
...

const res = await bulkPublishRelease(release.id,[publishActionPayload(3), publishActionPayload(4)])
```

Reply:
sounds good to me!